enhancements/cluster-logging/content-filter.md

```diff
@@ -0,0 +1,261 @@
+filters:
+  - name: important
+    type: drop
+    drop: # Type of filter, selets records to keep/drop
+      rules:
+      - field: .kubernetes.namespace_name
+        matches: so-important # Keep everything from this namespace regardless of level.
+        action: keep
+      - field: .level # Keep important levels
+        matches: warn(ing)?|err(or)?|critical
+        action: keep
+      - action: drop # Drop everything else.
+
+  - name: slim
+    prune: # Type of filter, prunes unwanted data
+      removeOthers: [.kubernetes, .message, .structured] # Remove all other fields.
+      remove: [.kubernetes.annotations] # Remove this sub-field.
+```
+
+##### Vector configuration
+
+See the implementation of the "kubeAPIAUdit" filter type, just the "rules" section.
+The tests are different but the iteration over a list of rules with actions is very similar.
+
+##### Pre-defined filters
+
+Just like default inputs ("application", "audit", "infrastructure") and outputs ("default"), we can define reserved names for pre-defined filters.
+These are equivalent to user-defined filters but pre-defined for convenience.
+
+- "DropDebug": `rules: { field: .level , in: [trace, debug] , action: drop }`
+- "DropInfo": `rules: { field: .level , in: [trace, debug, info] , action: drop }`
+
+
+### Implementation Details
+### JSON path
+
+Need to document the rules for field paths which are a subset of the JSON path spec.
+In fact we will use the same subset as Vector does, but we should describe the rules explicitly
+not refer to vector docs, so we don't create assumptions in the API about the underlying collector.
+
+### OTEL data model
+
+This proposal assumes log record paths are named according to the Viaq data model.
+
+Currently we are previewing an OTEL transformation at the HTTP output. That comes after the pipeline filters and does not affect this proposal.
+
+In future we may offer the OTEL data model as a "first class" data model, in place of Viaq.
+In that case, filter paths must be written according to the OTEL model.
+User filters using the Viaq model will need to be re-written, our pre-defined filters can be modifed to work with both models.
+
+### Risks and Mitigations
+
+- Performance impact, need to benchmark slowdown due to rule evaluation, but balance against reduction in data volume. 
+- No new security issues.
+- Extensions to ClusterLogForwarder will be reviewed as part of normal code review.
+
+### Drawbacks
+
+- Possible performance impact (negative and positive)
+- User can damage log records in ways that make them hard to display or process.
+
+## Design Details
+
+### Upgrade / Downgrade Strategy
+
+This is a new feature, opt-in and backward compatible. Should not cause any upgrade issues.
+
+### Version Skew Strategy
+
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+Invalid filters SHOULD be detected at reconcile time.
+Any errors at run time will cause a filter to be ignored, will not cause overall failure.
```

Comment:
Can we guarantee this?

Reply:
For the experience until now with Fluentd and Vector, when something is not able to be understood like special characters, it happened that the collector broke generating an stacktrace and stopping all. Then, it's very risky to say that will not cause overall failure